simulator/mips/mips_rf.h

```diff
@@ -82,9 +82,12 @@ class RF
 
         inline void write_dst( const FuncInstr& instr)
         {
-            RegNum reg_num = instr.get_dst_num();
-            if ( REG_NUM_ZERO != reg_num)
-                write( reg_num, instr.get_v_dst());
+            RegNum reg_num  = instr.get_dst_num();
+            uint32 v_dst    = instr.get_v_dst();
+            if ( REG_NUM_ZERO != reg_num && v_dst != NO_VAL32)
```

Comment:
You are preventing to write `0xdeadbeef` to registers, that's not good. An example:

```asm
or $t2, $zero, $zero
li $t1, 0xdeadbeef
or $t2, $t1, $t1; $t2 will contain 0x00000000
```

Reply:
Ok, I will think about it, can you please tell me, for each new operation v_dst is initialized with NO_VAL32, am  I right?

Reply:
And, as I understand your comment, in your example we must have NO_VAL32 in T1 register, but I'm preventing it, right?

Reply:
> for each new operation v_dst is initialized with NO_VAL32, am I right?

Yes, you may see it in code. We use it instead of "0" as "0xdeadbeff" should be a rare value and it is easier to understand its origin. However, there is no guarantee that such value is not used in traces.

> in your example we must have NO_VAL32 in T1 register, but I'm preventing it, right?

Yes. You have to think about better condition.

Reply:
Good afternoon! Can I just create a new constant like "0xdeadbeef" for this particular case, set it to v_dst if the condition was false and then check it in write_dst in the same way?

Reply:
No, it is the same thing. There is no warranty your "magic" constant won't be used.
The only was is to add a new Boolean variable (`writes_dst`) to FuncInstr and check its value with some method.